app/src/main/java/org/ole/planet/myplanet/ui/news/AdapterNews.java

```diff
@@ -125,12 +129,50 @@ private void addLabels(RecyclerView.ViewHolder holder, RealmNews news) {
             MenuInflater inflater = menu.getMenuInflater();
             inflater.inflate(R.menu.menu_add_label, menu.getMenu());
             menu.setOnMenuItemClickListener(menuItem -> {
-                if (!mRealm.isInTransaction())
-                    mRealm.beginTransaction();
-                news.addLabel(Constants.LABELS.get(menuItem.getTitle() + ""));
-                Utilities.toast(context, "Label added.");
-                mRealm.commitTransaction();
-                showChips(holder, news);
+                String menutitle = menuItem.getTitle().toString().toLowerCase();
+                Log.i("Menu Title", menutitle);
+                    if (!menutitle.equals("other")) {
+                        if (!mRealm.isInTransaction()) {
+                            mRealm.beginTransaction();
+                        }
+                        news.addLabel(Constants.LABELS.get(menuItem.getTitle() + ""));
+                        mRealm.commitTransaction();
+                        Utilities.toast(context, "Label added.");
+                        showChips(holder, news);
+                    } else {
+                        final String[] text = new String[1];
+                        View v = LayoutInflater.from(context).inflate(R.layout.alert_add_label, null);
```

Comment:
@anastasia21112   Why are you using ? use string instead of string array with length one.

Reply:
> @anastasia21112 Why are you using ? use string instead of string array with length one.

in a lambda expression, I can't change a string variable that was declared outside, so i would get an error. with an array, i didn't get an error.

Reply:
they require all variables to be final, and since you can't change a final String variable I decided to use an array instead. although i'm aware it's not the most efficient.

Reply:
@anastasia21112 then declare that variable inside setPositiveButton.
   alert.setPositiveButton("Save", (dialogInterface, i) -> {
                        String text = title.getText().toString();
                        Constants.LABELS.put(text, text);
...
...
...
}

Reply:
> @anastasia21112 then declare that variable inside setPositiveButton.
> 
> ```
>    alert.setPositiveButton("Save", (dialogInterface, i) -> {
>                         String text = title.getText().toString();
>                         Constants.LABELS.put(text, text);
> ...
> ...
> ...
> }
> ```

yes haha. i do not need to access it outside of the lambda. sorry, will fix that now!